packages/bundler/src/url-utils.ts

```diff
@@ -27,10 +27,11 @@ import constants from './constants';
  *     'file:///something/something.html_omg.js?ponies'
  */
 export function appendUrlPath(url_: string, extension: string): string {
-  const uri = Uri.parse(url_);
-  uri['_path'] = `${uri.path}${extension}`;
-  return uri.toString();
+  const {scheme, authority, path, query, fragment} = Uri.parse(url_);
```

Comment:
Is there a reason to list all the individual fields? How do we know we got all the important ones? I'd feel more confident with just updating the path...

Reply:
https://github.com/Microsoft/vscode-uri/blob/master/src/index.ts#L684

The reason to list them is that technically these are the only fields that should be in the input to `Uri.from()` whereas the Uri instance returned from a parse call contains more stuff according to typings and then we get compile errors.

Reply:
Can get around it with Object.assign but that's only because its a typings cheat.

Reply:
And object spread syntax returns error about unsupported fields like fsPath

Reply:
ok, makes sense if the types are different.